boards/ti/lp_mspm0g3507/lp_mspm0g3507_mspm0g3507_defconfig

```diff
@@ -0,0 +1,19 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_CORTEX_M_SYSTICK=y
+
+# Enable GPIO driver
+CONFIG_GPIO=y
+
+# Enable UART driver
+CONFIG_SERIAL=y
+
+# Enable Console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# Enable Pinctrl
+CONFIG_PINCTRL=y
+
```

Comment:
You will need to select this at soc level, not board level. 
See https://github.com/zephyrproject-rtos/zephyr/issues/78619

Reply:
I don't think this issue is suggesting SOC level, rather driver level. I'm seeing any driver that needs pinctrl is including it in their respective driver's Kconfig, i.e.

https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/Kconfig.nrfx#L12

Reply:
Agreed @JFMSP , this was a confusion from my end,  it looks like drivers need to enable this, boards can select in cases of board level mux which needs to be setup during initialization